docker/api/src/Domain/LineItem/LineItem.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/** @noinspection PhpPropertyOnlyWrittenInspection */
+
+namespace App\Domain\LineItem;
+
+use App\Domain\Product\Product;
+use App\Domain\User\User;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use JetBrains\PhpStorm\ArrayShape;
+use JsonSerializable;
+
+#[Entity, Table(name: 'lineItem')]
+class Line_Item implements JsonSerializable{
+
+    #[Id, Column(type: 'integer', unique: false, nullable: false) ]
+    private int $order_id;
+
+    #[Column(type: 'integer', unique: false, nullable: false)]
+    private int $product_id;
+
+    #[Column(type: 'float', unique: false, nullable: false)]
+    private int $lineItemPrice;
+
+    #[Column(type: 'integer', unique: false, nullable: false)]
+    private string $quantity;
+
+    #[ManyToOne(targetEntity: Product::class)]
+    #[JoinColumn(name: 'product_id', referencedColumnName: 'product_id')]
+    private Product $product;
+
+    // #[ManyToOne(targetEntity: Order::class)]
+    
```

Comment:
```suggestion
    public function setProduct(Product $product): void{
        $this->product = $product;
    }
```

Reply:
Without setting the product object, you will encounter an error "product_id cannot be null" when trying to add to database.  It seems all ManyToOne associations will throw this error, so the order object will also need to be set when that entity is completed.

Reply:
Great catch!  I already forgot about that.